docs/content/release-notes/flink-1.15.md

```diff
@@ -0,0 +1,597 @@
+`flink-connector-test-utils` module instead.
+
+#### Support partition keys through metadata (for FileSystem connector) 
+
+##### [FLINK-24617](https://issues.apache.org/jira/browse/FLINK-24617)
+
+Now the formats implementing `BulkWriterFormatFactory` don't need to implement
+partition keys reading anymore, as it's managed internally by `FileSystemTableSource`.
+
+#### Port ElasticSearch Sink to new Unified Sink API (FLIP-143) 
+
+##### [FLINK-24323](https://issues.apache.org/jira/browse/FLINK-24323)
+
+`ElasticsearchXSinkBuilder` supersedes `ElasticsearchSink.Builder` and provides at-least-once writing with the
+new unified sink interface supporting both batch and streaming mode of DataStream API.To upgrade, please stop with savepoint.
+
+For Elasticsearch 7 users that use the old ElasticsearchSink interface
+(`org.apache.flink.streaming.connectors.elasticsearch7.ElasticsearchSink`)
+and depend on their own elasticsearch-rest-high-level-client version,
+updating the client dependency to a version >= 7.14.0 is required due to internal changes.
+
+#### Reduce legacy in Table API connectors 
+
+##### [FLINK-24397](https://issues.apache.org/jira/browse/FLINK-24397)
+
+The old JDBC connector (indicated by `connector.type=jdbc` in DDL) has been removed.
+If not done already, users need to upgrade to the newer stack (indicated by `connector=jdbc` in DDL).
+
+## Runtime & Coordination
+
+#### Integrate retry strategy for cleanup stage 
+
+##### [FLINK-25433](https://issues.apache.org/jira/browse/FLINK-25433)
+
+Adds retry logic to the cleanup steps of a finished job. This feature changes the
+way Flink jobs are cleaned up. Instead of trying once to clean up the job, this
+step will be repeated until it succeeds. Users are meant to fix the issue that
+prevents Flink from finalizing the job cleanup. The retry functionality can be
+configured and disabled. More details can be found 
+[in the documentation](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/config/#retryable-cleanup).
+
+#### Introduce common interfaces for cleaning up local and global job data 
+
+##### [FLINK-25432](https://issues.apache.org/jira/browse/FLINK-25432)
+
+Utilizes the newly added component `JobResultStore` (see 
+[FLINK-25431](https://issues.apache.org/jira/browse/FLINK-25431)) to pick up
+already finished but not properly cleaned up jobs after a failover.
+
+#### Introduce explicit shutdown signaling between `TaskManager` and `JobManager` 
+
+##### [FLINK-25277](https://issues.apache.org/jira/browse/FLINK-25277)
+
+Reduce down-scaling delay in reactive mode through signaling between `TaskManager` 
+and `JobManager`.
+
+#### Release `TaskManagerJobMetricGroup` with the last slot rather than task 
+
+##### [FLINK-24864](https://issues.apache.org/jira/browse/FLINK-24864)
+
+TM-per-job metrics are now released upon last slot release rather than task, i.e. 
+may stay on TM longer.
```

Comment:
```suggestion
Job metrics on the TaskManager are now removed when the last slot is released, rather than the last task.
This means they may be reported for a longer time than before and when no tasks are running on the TaskManager.
```